src/jenkins/InputManifest.groovy

```diff
@@ -53,12 +53,35 @@ class InputManifest {
         }
     }
 
+    class Components extends HashMap<String, Component> {
+
+        Components(ArrayList data) {
+            data.each { item ->
+                Component component = new Component(item)
+                this[component.name] = component
+            }
+        }
+    }
+
+    class Component implements Serializable {
+        String name
+        String repository
+
+        Component(Map data) {
+            this.name = data.name
+            this.repository = data.repository
+        }
+
+    }
+
```

Comment:
Any reason not implementation the entire map like here?

https://github.com/opensearch-project/opensearch-build-libraries/blob/14bc0937325533f5541c53c9abfeac80b1e2c7df/src/jenkins/BundleManifest.groovy#L54-L72

Reply:
Currently `Components` Property of input Manifest is not used any where, so I have only added arguments which are used by the release-branch creation[ job](https://github.com/opensearch-project/opensearch-build/pull/4520)

Reply:
I do not agree with that approach.
If input manifest have those properties that you can use, just include them.
Bundle/Build manifest already have those why does input manifest be any different here? Unless it does not support those properties which is another case.

Reply:
Sure I can do that. Thanks Peter